smartapps/initialstate-events/initial-state-event-streamer.src/initial-state-event-streamer.groovy

```diff
@@ -330,57 +330,53 @@ def genericHandler(evt) {
 // This is a handler function for flushing the event buffer
 // after a specified amount of time to reduce the load on ST servers
 def flushBuffer() {
+	def eventBuffer = atomicState.eventBuffer
 	log.trace "About to flush the buffer on schedule"
-	if (atomicState.eventBuffer != null && atomicState.eventBuffer.size() > 0) {
-		tryShipEvents()
+	if (eventBuffer != null && eventBuffer.size() > 0) {
+		atomicState.eventBuffer = []
```

Comment:
Clear the eventBuffer here shortly after we retrieve it so that we don't miss events between now and when we actually post the current eventBuffer to Initial State.